src/components/Payroll/PayrollLanding/PayrollLanding.tsx

```diff
@@ -21,21 +24,58 @@ export function PayrollLanding(props: PayrollLandingProps) {
 
 export const Root = ({ onEvent, companyId }: PayrollLandingProps) => {
   const [selectedTab, setSelectedTab] = useState('run-payroll')
+  const [showReceiptForPayrollId, setShowReceiptForPayrollId] = useState<string | null>(null)
+  const [showSummaryForPayrollId, setShowSummaryForPayrollId] = useState<string | null>(null)
   const { Tabs } = useComponentContext()
 
   useI18n('Payroll.PayrollLanding')
   const { t } = useTranslation('Payroll.PayrollLanding')
 
+  const wrappedOnEvent: typeof onEvent = (event, payload) => {
```

Comment:
awesome, I'm gonna borrow this command style pattern down the road when deal with components that might emit similar events